test/cluster-setup.sh

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+
+# Copyright 2018 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script calls out to scripts in knative/test-infra to setup a cluster
+# and deploy the Pipeline CRD to it for running integration tests.
+
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
+
+set -o xtrace
+set -o errexit
+set -o pipefail
+
+function take_down_pipeline() {
+    header "Tearing down Pipeline CRD"
+    ko delete --ignore-not-found=true -f config/
+}
+trap take_down_pipeline EXIT
+
+# Called by `fail_test` (provided by `e2e-tests.sh`) to dump info on test failure
+function dump_extra_cluster_state() {
+  for crd in pipelines pipelineruns tasks taskruns resources pipelineparams
+  do
+    echo ">>> $crd:"
+    kubectl get $crd -o yaml --all-namespaces
+  done
+  echo ">>> Pipeline controller log:"
+  kubectl -n knative-build-pipeline logs $(kubectl -n knative-build-pipeline get pods -l app=build-pipeline-controller -o name)
+  echo ">>> Pipeline webhook log:"
+  kubectl -n knative-build-pipeline logs $(kubectl -n knative-build-pipeline get pods -l app=build-pipeline-webhook -o name)
+}
+
+set +o xtrace
+header "Setting up environment"
+# The intialize method will attempt to create a new cluster in $PROJECT_ID unless
+# the `--run-tests` parameter is provided and `K8S_CLUSTER_OVERRIDE` is set, however
+# since knative/test-infra/scripts/presubmit-tests.sh doesn't propagate `--run-tests`
+# (and it's kind of a confusing param name), we'll infer it from the presence of
+# `K8S_CLUSTER_OVERRIDE`
+if [[ -z ${K8S_CLUSTER_OVERRIDE} ]]; then
```

Comment:
You don't need this. This is all handled by the helper.
See the most simple E2E script there is:

https://github.com/knative/test-infra/blob/master/test/e2e-tests.sh

It's so simple that the number of comment lines is much greater than the number of actual code lines. :D

The header of the helper should contain all the instructions about writing your own script.

Reply:
How do I work around needing `--run-tests`?

Reply:
You don't need `--run-tests`, unless you want to run your tests against an already up-and-running cluster. In this case you need the environment vars: https://github.com/knative/test-infra/tree/master/scripts#using-the-e2e-testssh-helper-script (note 2).

Reply:
hm but when using an already created cluster, I need to provide `--run-tests` so that `e2e-tests.sh` [will setup a cluster instead of creating one](https://github.com/knative/test-infra/blob/master/scripts/e2e-tests.sh#L302):

![image](https://user-images.githubusercontent.com/432502/46438581-21ca1d00-c713-11e8-8659-e5119c19627f.png)

Reply:
but using `presubmit-tests.sh` prevents me from being able to pass `--run-tests` since [it isn't a recognized argument](https://github.com/knative/test-infra/blob/master/scripts/presubmit-tests.sh#L83)

Reply:
You don't provide `--run-tests` to `e2e-tests.sh` (unless it's from command line), it will take care of it itself.